File: cinematch-backend-service/cinematch-backend/src/main/java/fr/cpe/cinematch_backend/dtos/requests/MovieSearchRequest.java
package fr.cpe.cinematch_backend.dtos.requests;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
public class MovieSearchRequest {
    private String title;
    private List<String> genres;
    private List<String> director;
    private List<String> cast;
    private Float minRating;
    private Float maxRating;
    private Date startDate;
    private Date endDate;
}

File: cinematch-frontend/src/models/movie-search-request.ts
export class MovieSearchRequest {
    title: string | null;
    genres: string[] | null;
    director: string[] | null;
    cast: string[] | null;
    minRating: number | null;
    maxRating: number | null = 10;
    startDate: Date | null;
    endDate: Date | null;
}
